python/python/lancedb/db.py

```diff
@@ -886,15 +886,23 @@ async def rename_table(self, old_name: str, new_name: str):
         """
         await self._inner.rename_table(old_name, new_name)
 
-    async def drop_table(self, name: str):
+    async def drop_table(self, name: str, ignore_missing: bool = False):
```

Comment:
Also, let's make this kwarg-only in the async API:
```suggestion
    async def drop_table(self, name: str, *, ignore_missing: bool = False):
```